submissions/Siusarna/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,141 @@
+const net = require('net');
+const dns = require('dns').promises;
+
+const args = process.argv;
+const help = `
+  Simple port sniffer
+This script takes 2 parameters:
+
+    --host     obligatory parameter which defines the host address for a port scan.
+               Address must be specified after the parameter.
+               It can be specified as IP address or URL.
+
+    --ports     Parameter which specifies port range.
+               After the parameter two numbers from 0 to 65535 must be specified with a delimiter '-' between them.
+
+               Example: node sniffer.js --host google.com --port 100-2345
+
+    --help     Parameter will show this instructions.
+`;
+
+const validHost = async host => {
+  if (!host) {
+    process.stdout.write(
+      'Invalid host parameter. Try node sniffer.js --help \n'
+    );
+  }
+  try {
+    return await dns
+      .lookup(host)
+      .then(res => {
+        return res.address;
+      })
+      .catch(error => {
+        process.stdout.write(
+          `Invalit host name. ${error}. Try node sniffer.js --help \n`
+        );
+        return 0;
+      });
+  } catch (e) {
+    process.stdout.write(e);
+    return 0;
+  }
+};
+
+const checkRangePort = port => {
+  if (port[0] < 0 || port[1] > 65535 || port[2] || port[0] > port[1]) {
+    process.stdout.write('Invalid port range. See help \n');
+    return false;
+  }
+  return true;
+};
+
+const validatorForPort = port => {
+  let validPort = [1, 65535];
+  if (port && checkRangePort(port)) {
+    validPort = port;
+  } else {
+    process.stdout.write(
+      'No port range specified. Setting to default. For more information try --help \n'
+    );
+  }
+  return validPort;
+};
+
+const checkConnection = async (port, host) => {
+  let flag = false;
+  await new Promise((resolve, reject) => {
+    const socket = new net.Socket();
+    socket.setTimeout(300);
+    socket.on('connect', () => {
+      process.stdout.write('.');
+      flag = true;
+      socket.destroy();
+      resolve();
+    });
+    socket.on('timeout', () => {
+      socket.destroy();
+      reject();
+    });
+    socket.on('error', () => {
+      socket.destroy();
+      reject();
+    });
+    socket.connect(port, host);
+  }).catch(e => {
+    if (e !== undefined) process.stdout.write(e);
```

Comment:
I don't think `e` is optional here. Couldn't find any case when error object would be `indefined`